src/client/datascience/types.ts

```diff
@@ -404,13 +404,29 @@ export interface IInteractiveWindowProvider {
     get(owner: Uri): IInteractiveWindow | undefined;
 }
 
+type CanceledKernelErrorResult = {
+    kind: 'Canceled';
+};
+type ErrorKernelErrorResult = {
+    kind: 'Error';
+    error: Error;
+};
+type InstalledKernelErrorResult = {
+    kind: 'Installed';
+};
+type SwitchedKernelErrorResult = {
+    kind: 'Switched';
```

Comment:
Can we make this more verbosed like `SwitchKernel` `InstalledDependencies` `UnhandledError`
I'm assuming kind=`Error` stands for `UnhandledError`, i.e. we do'nt know what to do about that.